File: ios/KhrysalisTemplate/com/lightningkite/khrysalistemplate/vg/MultipleDemoVG.shared.swift
//Package: com.lightningkite.khrysalistemplate.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps
import KhrysalisBluetooth



public class MultipleDemoVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Multiple List Demo"
        }
    }
    public var data: Array<Any>
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = MultipleDemoXml()
        var view = xml.setup(dependency)
        xml.list.bindMulti(viewDependency: dependency, data: ConstantObservableProperty(data), typeHandlerSetup: { (handler) in 
            handler.handle(0.toInt()){ (obs) in 
                var cellXml = ComponentTestXml()
                var cellView = cellXml.setup(dependency)
                cellXml.label.bindString(obs.map{ (it) in 
                    "The number \(it)"
                })
                return cellView
            }
            handler.handle(""){ (obs) in 
                var cellXml = ComponentTestXml()
                var cellView = cellXml.setup(dependency)
                cellXml.label.bindString(obs.map{ (it) in 
                    "The string '\(it)'"
                })
                return cellView
            }
            handler.handle(()){ (obs) in 
                var cellXml = ComponentTextXml()
                var cellView = cellXml.setup(dependency)
                cellXml.label.text = "-----"
                return cellView
            }
        })
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        let data: Array<Any> = [1.toInt(), 2.toInt(), "3", 4.toInt(), "5", (), "Test", 6.toInt()] as! Array<Any>
        self.data = data
        super.init()
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/vg/MultipleDemoVG.shared.kt
package com.lightningkite.khrysalistemplate.vg

import android.view.View
import com.lightningkite.khrysalis.observables.binding.bindMulti
import com.lightningkite.khrysalis.observables.binding.bindString
import com.lightningkite.khrysalis.observables.ConstantObservableProperty
import com.lightningkite.khrysalis.observables.map
import com.lightningkite.khrysalis.views.ViewDependency
import com.lightningkite.khrysalis.views.ViewGenerator
import com.lightningkite.khrysalistemplate.layouts.*

class MultipleDemoVG() : ViewGenerator() {
    override val title: String get() = "Multiple List Demo"

    val data: List<Any> = listOf(
        1.toInt(),
        2.toInt(),
        "3",
        4.toInt(),
        "5",
        Unit,
        "Test",
        6.toInt()
    ) as List<Any>

    override fun generate(dependency: ViewDependency): View {
        val xml = MultipleDemoXml()
        val view = xml.setup(dependency)

        xml.list.bindMulti(
            viewDependency = dependency,
            data = ConstantObservableProperty(data),
            typeHandlerSetup = { handler ->
                handler.handle(0.toInt()) { obs ->
                    val cellXml = ComponentTestXml()
                    val cellView = cellXml.setup(dependency)
                    cellXml.label.bindString(obs.map { it ->"The number $it" })
                    return@handle cellView
                }
                handler.handle("") { obs ->
                    val cellXml = ComponentTestXml()
                    val cellView = cellXml.setup(dependency)
                    cellXml.label.bindString(obs.map { it ->"The string '$it'" })
                    return@handle cellView
                }
                handler.handle(Unit) { obs ->
                    val cellXml = ComponentTextXml()
                    val cellView = cellXml.setup(dependency)
                    cellXml.label.text = "-----"
                    return@handle cellView
                }
            }
        )

        return view
    }
}
